services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/settings.py

```diff
@@ -31,6 +41,10 @@ class ResourceTrackingSettings(BaseCustomSettings):
         description="each time the status of the service is propagated",
     )
 
+    _legacy_parsing_dynamic_scheduler_stop_service_timeout = (
+        validate_timedelta_in_legacy_mode("RESOURCE_TRACKING_HEARTBEAT_INTERVAL")
```

Comment:
did you check this: https://docs.pydantic.dev/latest/api/standard_library_types/#datetimetimedelta
the documentation states that it still works correctly.

Reply:
The issue is that we were providing str timedeltas in env vars with a different format (the number of seconds). Now this is no longer accepted, but it used to work with pydantic1

Reply:
so the naming is really weird. I think we do not care that it might be legacy or so. this should become a real validator converting a numeric string into seconds.

Reply:
alright, so what I did for the autoscaling, is go away from that float/int nonsense.
I changed them to follow ISO 8601: "00:00:02" if you want 2 seconds. and that is it

Reply:
we also have to convert all the settings int env vars repos to ensure this is still working. And I did not want to mess up also with those, that's the reason for this validator.
let me rename it a bit and see if it's better, at least

Reply:
well it's a migration.. and there are many of these interval where the validator was not setup.

Reply:
applied everywhere and added a nice warning message
```
tests/unit/test_core_settings.py::test_settings_with_envdevel_file
  /home/silenthk/work/pr-osparc-pydantic-v2-dynamic-sidecar/services/dynamic-sidecar/tests/unit/test_core_settings.py:15: DeprecationWarning: RESOURCE_TRACKING_HEARTBEAT_INTERVAL='33444' --should be set to--> RESOURCE_TRACKING_HEARTBEAT_INTERVAL='PT9H17M24S' (ISO8601 datetime format). Please also convert the value in the >>OPS REPOSITORY<<. For details: https://docs.pydantic.dev/1.10/usage/types/#datetime-types.
    assert ApplicationSettings.create_from_envs()
```